pkg/webhook/resources/version/validator.go

```diff
@@ -55,5 +60,25 @@ func (v *versionValidator) checkAnnotations(version *v1beta1.Version) error {
 
 func (v *versionValidator) Update(_ *types.Request, _ runtime.Object, newObj runtime.Object) error {
 	newVersion := newObj.(*v1beta1.Version)
-	return v.checkAnnotations(newVersion)
+	return checkVersion(newVersion)
+}
+
+func checkVersion(version *v1beta1.Version) error {
+	if err := checkAnnotations(version); err != nil {
+		return err
+	}
+	if err := checkISOChecksum(version); err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkISOChecksum(version *v1beta1.Version) error {
+	isoChecksum := version.Spec.ISOChecksum
+	// if an isoChecksum is provided, it must be in the SHA-512 format
+	// since Longhorn backing images only accept hashes in that format
+	if isoChecksum != "" && !SHA512Pattern.MatchString(isoChecksum) {
+		return werror.NewBadRequest(fmt.Sprintf("invalid isoChecksum %s", isoChecksum))
```

Comment:
The error message could hint that the ` the provided isoChecksum is not in the expected SHA512 format`

And other question:

If the `isoCheckSum` is in correct format, but the value is wrong, will we have the exact error message from LH to show to user?

Reply:
Yes, we'll have something like `Failed to process sync file: the expected checksum ... doesn't match the file actual checksum ...` in both upgrade and image download page.